### PR TITLE
Fix unregistering pusher local error

### LINKED ISSUE #2895: Selecting ntfy as notification provider does not work
### Steps to reproduce

1. In "Advanced settings", tap "Push notification provider"
2. From the selection of "Firebase" and "ntfy", pick "ntfy"
3. "Push notification provider" setting now shows "Error" as current value, going back and then into "Advanced settings" again makes it switch back to "Firebase"

#### What did you expect?

It is possible to use ntfy as a push notification provider.

#### What happened instead?

Changing the provider to ntfy fails. Other apps like FediLab can register their ntfy topics without issue.

### Your phone model

Samsung S10

### Operating system version

LineageOS for microG 20

### Application version and app store

EXA 0.4.13

### Homeserver

xiretza.xyz, Synapse v1.105.1

### Will you send logs?

Yes

### Are you willing to provide a PR?

No

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Fixes #2895 by ignoring local error when unregistering a pusher.

Base on the `unitTestPush` branch to be able to update the test, but the other PR will be merged first.

## Motivation and context

Fixes #2895 

## Screenshots / GIFs

NA

## Tests

- Hard to test without hacking the code, but IRL this seems to happen.

Maybe @Xiretza may want to test the patch.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

**Review by jmartinesp — APPROVED** [review]
The code looks good, I just have a nitpick. However, I couldn't get NTFY working (maybe `matrix.org` is still rate limited?) so I couldn't really test it. I'll trust you on this one 👍 .

File: libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/DefaultPushService.kt [hunk 1]
```diff
@@ -51,13 +51,16 @@ class DefaultPushService @Inject constructor(
...
+        Timber.d("Registering with ${pushProvider.name}/${distributor.name}}")
         val userPushStore = userPushStoreFactory.getOrCreate(matrixClient.sessionId)
         val currentPushProviderName = userPushStore.getPushProviderName()
         val currentPushProvider = pushProviders.find { it.name == currentPushProviderName }
         val currentDistributorValue = currentPushProvider?.getCurrentDistributor(matrixClient)?.value
         if (currentPushProviderName != pushProvider.name || currentDistributorValue != distributor.value) {
             // Unregister previous one if any
-            currentPushProvider?.unregister(matrixClient)
+            currentPushProvider
+                ?.also { Timber.d("Unregistering previous push provider $currentPushProviderName/$currentDistributorValue") }
```
[the remark this answers is not in the document]
Comment (bmarty) [reply libraries/push/impl/src/main/kotlin/io/element/android/libraries/push/impl/DefaultPushService.kt:62]: Yeah, not really since here `currentPushProvider` can still be null, so here `?.` for the `also`  is useful.
I agree that the code is a bit hard to follow, I tried to add some `if(x (!= null)` statement but it's getting worse.
